File: src/Java/172.TrailingZeroes.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;
import java.util.Set;
import java.util.HashSet;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.Map;

class Solution {
    public int trailingZeroes(int n) {
        int sum = 0;
        for (long i = 5; i <= n; i *= 5) {
            sum += n / i;
        }
        return sum;
    }
}

File: src/Python/172.TrailingZeroes.py

class Solution:
    def trailingZeroes(self, n):
        """
        :type n: int
        :rtype: int
        """
        count = 0
        while True:
            n = n // 5
            if n == 0:
                break
            else:
                count += n
        return count

if __name__ == '__main__':
    solution = Solution()
    for i in range(101):
        print(solution.trailingZeroes(i))
else:
    pass
